src/cli/cli.cc

```diff
@@ -0,0 +1,229 @@
+        uv_close((uv_handle_t *)&client_pipe, NULL);
+        return;
+    }
+}
+
+static void pipe_write_cb(uv_write_t *req, int status)
+{
+    int ret;
+    (void)status;
+
+    uv_pipe_t *clientp = (uv_pipe_t *)req->data;
+    shutdown_req.data = clientp->data;
+
+    ret = uv_shutdown(&shutdown_req, (uv_stream_t *)&client_pipe, shutdown_cb);
+    if (ret) {
+        fprintf(stderr, "uv_shutdown(): %s\n", uv_strerror(ret));
+        uv_close((uv_handle_t *)&client_pipe, NULL);
+        return;
+    }
+}
+
+static void connect_cb(uv_connect_t *req, int status)
+{
+    int ret;
+    uv_buf_t write_buf;
+
+    if (status) {
+        fprintf(stderr, "uv_pipe_connect(): %s\n", uv_strerror(status));
+        fprintf(stderr, "Make sure the netdata service is running.\n");
+        exit(-1);
+    }
+
+    client_pipe.data = req->data;
+    write_req.data = &client_pipe;
+    write_buf.base = command_string;
+    write_buf.len = command_string_size;
+    ret = uv_write(&write_req, (uv_stream_t *)&client_pipe, &write_buf, 1, pipe_write_cb);
+    if (ret) {
+        fprintf(stderr, "uv_write(): %s\n", uv_strerror(ret));
+    }
+}
+
+int main(int argc, char **argv)
+{
+    nd_log_initialize_for_external_plugins("netdatacli");
+
+    CLI::App app{"Netdata CLI Tool"};
+
+    // Disable default help flag since we'll handle it ourselves
+    app.set_help_flag("");
+
+    // Variables to store option states
+    bool reload_health = false;
+    bool reload_labels = false;
+    bool reopen_logs = false;
+    bool shutdown_agent = false;
+    bool fatal_agent = false;
+    bool reload_claiming_state = false;
+    bool ping = false;
+    bool dumpconfig = false;
+    bool version = false;
+    std::string aclk_state;
+    std::string remove_stale_node;
+
+    app.set_help_flag("--help", "Show this help message and exit");
+    app.add_flag("--reload-health", reload_health, "Reload health configuration");
+    app.add_flag("--reload-labels", reload_labels, "Reload all labels");
+    app.add_flag("--reopen-logs", reopen_logs, "Close and reopen log files");
+    app.add_flag("--shutdown-agent", shutdown_agent, "Cleanup and exit the netdata agent");
+    app.add_flag("--fatal-agent", fatal_agent, "Log the state and halt the netdata agent");
+    app.add_flag("--reload-claiming-state", reload_claiming_state, "Reload agent claiming state from disk");
+    app.add_flag("--ping", ping, "Return with 'pong' if agent is alive");
+    app.add_option(
+        "--aclk-state", aclk_state, "Returns current state of ACLK and Cloud connection. Use 'json' for JSON format");
+    app.add_flag("--dump-config", dumpconfig, "Returns the current netdata.conf on stdout");
+    app.add_option(
+        "--remove-stale-node",
+        remove_stale_node,
+        "Unregisters and removes a node from the cloud. Specify node_id, machine_guid, hostname, or ALL_NODES");
+    app.add_flag("--version", version, "Returns the netdata version");
```

Comment:
Changing all of these to use a `--` prefix is a breaking change. Ideally we need to keep the old syntax without the `--` to avoid breaking things (though I would like to see `--help` supported alongside `help`, and probably ideally also a `--version` option since we should be able to provide that).